src/promptflow/tests/sdk_cli_test/recording_utilities/mocked_functions.py

```diff
@@ -0,0 +1,175 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import os
+from functools import partial
+from pathlib import Path
+from typing import Union
+
+from promptflow._sdk._constants import ConnectionType
+from promptflow._sdk._errors import ConnectionNotFoundError
+from promptflow._sdk.operations._local_storage_operations import NodeRunRecord
+from promptflow._sdk.operations._test_submitter import TestSubmitter
+from promptflow._utils.tool_utils import get_inputs_for_prompt_template
+from promptflow.contracts.flow import Node, ToolSourceType
+from promptflow.contracts.run_info import RunInfo as NodeRunInfo
+from promptflow.exceptions import ErrorTarget, PromptflowException, UserErrorException
+from promptflow.executor._errors import ResolveToolError
+from promptflow.executor._tool_resolver import ResolvedTool, ToolType
+from promptflow.executor.flow_executor import LineResult
+
+from .tool_record import just_return, record_node_run
+
+
+def mock_update_run_func(self, run_info):
+    run_id = run_info.run_id
+    run_info.api_calls = self._collect_traces_from_nodes(run_id)
+    child_run_infos = self.collect_child_node_runs(run_id)
+    run_info.system_metrics = run_info.system_metrics or {}
+    run_info.system_metrics.update(self.collect_metrics(child_run_infos, self.OPENAI_AGGREGATE_METRICS))
+    run_info.system_metrics["total_tokens"] = 0
+
+
+def mock_persist_node_run(recording_folder: Path):
+    def _mock_persist_node_run(self, run_info: NodeRunInfo) -> None:
```

Comment:
I suppose this mock is kind of big because you copy some original code; maybe you can add some comment on what is the key point for this mock?